docs/operator.md

```diff
@@ -0,0 +1,117 @@
+
+The relevant resource yamls are then copied to ./config/assisted-service where additional customizations are applied using Kustomize. The resulting yaml is then piped to operator-sdk
+creating the manifests and CSVs in ./bundle/manifests. 
+
+More information about bundles: <https://sdk.operatorframework.io/docs/olm-integration/generation/>
+
+### Create the bundle and index images
+
+```bash
+export ORG=quay.io/ocpmetal
+make operator-bundle-build BUNDLE_IMG=$ORG/assisted-service-operator-bundle:0.0.1
+podman push $ORG/assisted-service-operator-bundle:0.0.1
+
+opm index add --bundles $ORG/assisted-service-operator-bundle:0.0.1 --tag $ORG/assisted-service-index:0.0.1
+podman push $ORG/assisted-service-index:0.0.1
+```
+
+## Deploying the operator through OperatorHub
+
+The operator must be deployed to assisted-installer namespace. Create the namespace. 
+
+```bash
+cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: assisted-installer
+  labels:
+    name: assisted-installer
+EOF
+```
+
+Create two PVCs, one for postgres and another for s3 (temporary until filesystem implementation is available).
+
+```bash
+cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  labels:
+    app: postgres
+  name: postgres-pv-claim
+  namespace: assisted-installer
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+EOF
+
+cat <<EOF | kubectl create -f -
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  labels:
+    app: scality
+  name: scality-pv-claim
+  namespace: assisted-installer
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 20Gi
+EOF
+```
+
+Create a catalog source for the operator to appear in OperatorHub.
+
+``` bash
+cat <<EOF | kubectl create -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: assisted-service
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: quay.io/ocpmetal/assisted-service-index:0.0.1
+EOF
+```
+
+It may take a few minutes for the operator to appear in Operatorhub.
+
+## Useful Kustomize options
+
+In [`config/default/kustomization.yaml`](https://github.com/openshift/assisted-service/blob/master/config/default/kustomization.yaml):
+
+```
+# Uncomment to set a mininum disk size allowed by the hardware validator.
+# By default the minimum disk size allowed is 120GB. The patch sets the
+# minimum disk size to 20GB.
+#- assisted-service-configmap-patch-hw-validator-min-disk-size.yaml
+```
+
+```
+# Uncomment to use a custom assisted-service image in the deployment
+#- assisted-service-patch-image.yaml
+```
```

Comment:
Hey @rwsu I would add some Sample CR's in order to let the users try it out, at least the ClusterDeployment and the InstallEnv ones. This way you will help them to test it faster.

Reply:
+1 
Can probably done in a different PR